### PR TITLE
Add support for DynamoDB Backup/Restore

### PR DESCRIPTION
**Pull request by bpandola**
Basic support for the following endpoints has been implemented with full test coverage:
- create_backup
- delete_backup
- describe_backup
- list_backups
- restore_table_from_backup

Behavior and error messages verified against a real AWS backend.

### CODE REVIEW

**Comment by bpandola** [comment]
@bblommers Looks like another flaky test... I see other PRs with the same failure (`FAILED tests/test_elbv2/test_elbv2.py::test_handle_listener_rules`).

I'll have a look to see if it's an easy fix.

File: tests/test_dynamodb2/test_dynamodb.py [hunk 1]
```diff
@@ -6008,3 +6008,302 @@ def test_get_item_for_non_existent_table_raises_error():
...
+    details = backup_deleted["BackupDetails"]
+    details.should.have.key("BackupArn").should.equal(backup_to_delete)
+    details.should.have.key("BackupName").should.be.within(backup_names)
+    details.should.have.key("BackupStatus").should.equal("DELETED")
+    resp = client.list_backups(TableName=table_name, BackupType="USER")
+    resp["BackupSummaries"].should.have.length_of(1)
+
+
+@mock_dynamodb2
+def test_source_and_restored_table_items_are_not_linked():
```
Comment (bblommers) [inline tests/test_dynamodb2/test_dynamodb.py:6265]: Can we extend this test a bit, to verify the behaviour for items created between backup-creation and restoring?
```
create_backup
add_items to original table
restore_backup
verify original has x items
verify backup only has y items
```

**Review by bblommers — CHANGES_REQUESTED** [review]
Thanks for the PR @bpandola! Just one comment, but happy with how it looks overall.

I'll have a look at the failing test as well later today.

**Review by bblommers — APPROVED** [review]
LGTM!

**Pull request merged**